src/Components/QuickGrid/Microsoft.AspNetCore.Components.QuickGrid/test/NotFailingQuickGrid.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components.QuickGrid.Tests;
+/// <summary>
+/// A QuickGrid implementation that uses the same implementation of the basic QuickGrid with the additions of the OnAfterRenderCompleted task.
+/// </summary>
+/// /// <typeparam name="TGridItem">The type of data represented by each row in the grid.</typeparam>
+internal class NotFailingGrid<TGridItem> : QuickGrid<TGridItem>
+{
+    [Inject] private IJSRuntime JS { get; set; } = default!;
+
+    private readonly TaskCompletionSource _onAfterRenderCompleted = new();
+
+    private bool _completionSignaled;
+
+    /// <summary>
+    /// Task that completes when OnAfterRenderAsync has finished executing.
+    /// This allows tests to wait deterministically for the race condition to occur.
+    /// </summary>
+    public Task OnAfterRenderCompleted => _onAfterRenderCompleted.Task;
+
+    protected override async Task OnAfterRenderAsync(bool firstRender)
+    {
+        try
+        {
+            if (firstRender)
+            {
+                await base.OnAfterRenderAsync(firstRender);
+                if (!_completionSignaled)
```

Comment:
This is a duplicate of finally action. If we throw in `OnAfterRenderAsync` -> we go to finally. If we don't throw -> we execute this condition and by default flow, we should go to finally. `return` is redundant.

Remove this one and use `finally` only. We have the same construction in `FailingQuickGrid`.